### PR TITLE
Command-line (rake task) PDF batch ingest

### PR DESCRIPTION
**Pull request by seanupton**
Implements functionality described in #95, plus publication metadata retrieval from upstream sources (lccn.loc.gov, ChronAm) used by both PDF and NDNP ingests.

### CODE REVIEW

File: lib/newspaper_works/ingest/pub_finder.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+      def find_or_create_publication_for_issue(issue, lccn, title, opts)
+        publication = find_publication(lccn)
+        unless publication.nil?
+          write_log(
+            "Found existing NewspaperTitle #{publication.id}, LCCN #{lccn}"
+          )
+        end
+        publication = create_publication(lccn, title, opts) if publication.nil?
+        publication.ordered_members << issue
```
Comment (ebenenglish) [inline lib/newspaper_works/ingest/pub_finder.rb]: Do we want to use `ordered_members` here? Have we used this in the past for NDNP? My recollection is that issues are related to publications by `members` but not `ordered_members`.

File: spec/lib/newspaper_works/ingest/pdf_issue_ingester_spec.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    it "ingests PDFs" do
+      ingester = described_class.new(single_issue_dir)
+      ingester.ingest
+      # Outcomes tested:
+      # 1. NewspaperTitle fpr Publication created
+      publication = NewspaperTitle.where(lccn: lccn).first
+      expect(publication).not_to be_nil
+      # 2. NewspaperIssue created and findable withing publication members
+      issue = publication.ordered_members.to_a[0]
```
Comment (ebenenglish) [inline spec/lib/newspaper_works/ingest/pdf_issue_ingester_spec.rb]: Same comment here about `ordered_members`.

File: spec/lib/newspaper_works/ingest/pdf_issue_ingester_spec.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      expect(issue.publication_date).to eq "1853-06-04"
+      puts(publication.ordered_members.to_a.size)
```
Comment (ebenenglish) [inline spec/lib/newspaper_works/ingest/pdf_issue_ingester_spec.rb]: Guessing this is a debug line that should be removed?

File: lib/newspaper_works/ingest/pub_finder.rb (hunk 1 above)
Comment (seanupton) [reply lib/newspaper_works/ingest/pub_finder.rb]: This is a good question.  This is a holdover from how it was done with NDNP ingest, but there is  not a reason I can think of to need the ordering.

I suspect that using members is reasonable — the only frustration I have with these associations is that `ordered_members` and `members` are wholly disjoint.  Ideally, there would be an accessor that allowed access to the combined set (vs. cognitive load of having to remembers to check each, both, or one or the other).

File: spec/lib/newspaper_works/ingest/pdf_issue_ingester_spec.rb (hunk 2 above)
Comment (seanupton) [reply spec/lib/newspaper_works/ingest/pdf_issue_ingester_spec.rb]: Agreed, see note above.  I think I will change this in both pub_finder (setting) and here (access).

File: spec/lib/newspaper_works/ingest/pdf_issue_ingester_spec.rb (hunk 3 above)
Comment (seanupton) [reply spec/lib/newspaper_works/ingest/pdf_issue_ingester_spec.rb]: Yes, will remove.

File: lib/newspaper_works/ingest/pub_finder.rb (hunk 1 above)
Comment (seanupton) [reply lib/newspaper_works/ingest/pub_finder.rb]: This and other feedback should be resolved by commit 208438424f64de1ac9900c6be36b1a918f29494f

File: lib/newspaper_works/ingest/chronam_publication_info.rb [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            "\tlccn: '#{@lccn}'>",
+          oid: object_id << 1
+        )
+      end
+
+      def load_place
+        place_match = find('//rda:placeOfPublication')
+        return if place_match.nil?
+        @place_name = place_match.first.text
+        @place_of_publication = NewspaperWorks::Ingest.geonames_place_uri(
```
Comment (ebenenglish) [inline lib/newspaper_works/ingest/chronam_publication_info.rb]: For indexing to work as expected, the `place_of_publication` value should be set as a `Hyrax::ControlledVocabularies::Location` object, not `String.` Something like this should work:
```
def load_place
  place_match = find('//rda:placeOfPublication')
  return if place_match.nil?
  @place_name = place_match.first.text
  uri = NewspaperWorks::Ingest.geonames_place_uri(@place_name)
  @place_of_publication = Hyrax::ControlledVocabularies::Location.new(uri)
end
```
Specs will also need to be adjusted accordingly.

File: lib/newspaper_works/ingest/lc_publication_info.rb [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        return "#{city.text}, #{state.text}" if city && state
+        # fallback to placeTerm text, which may be abbreviated in such a
+        #   way that geonames struggles to find on search; for a list of
+        #   abbreviations, see:
+        #   https://www.loc.gov/aba/publications/FreeSHM/H0810.pdf
+        name = find('//mods:originInfo//mods:placeTerm[@type="text"]').first
+        name.nil? ? nil : name.text
+      end
+
+      def load_place
```
Comment (ebenenglish) [inline lib/newspaper_works/ingest/lc_publication_info.rb]: Same comment as with `ChronAmPublicationInfo#load_place`. The value of `@place_of_publication` needs to be `Hyrax::ControlledVocabularies::Location.new(uri)`.

**Pull request merged**